editor/utils/heartbeat-post-locking.js

```diff
@@ -0,0 +1,83 @@
+/* global ajaxurl */
+
+/**
+ * WordPress dependencies
+ */
+import { dispatch, select } from '@wordpress/data';
+
+export function setupHearthbeatPostLocking() {
+	/**
+	 * Configure Heartbeat post locks.
+	 *
+	 * Used to lock editing of an object by only one user at a time.
+	 *
+	 * When the user does not send a heartbeat in a heartbeat-tick
+	 * the user is no longer editing and another user can start editing.
+	 */
+	jQuery( document ).on( 'heartbeat-send.refresh-lock', function( e, data ) {
```

Comment:
Should we import `jQuery` instead of using the global (since it's already aliased)?

Midterm/longterm, should we rewrite heartbeat as a package leveraging?

 - hooks for these events
 - drop jQuery
 - use apiFetch

Reply:
Minor: In general, I prefer to avoid `functions` as closures and always favor arrow functions in that case.

Reply:
arrow functions: good point, i'll switch this to an arrow function. The JS for this file originated in core's post locking so I brought over the vanilla JS approach.

dropping jquery: love that idea or rewriting using hooks long term. since we need to support locking across classic/gutenberg we may not be able to this until post merge.